exec/java-exec/src/test/java/org/apache/drill/TestPartitionFilter.java

```diff
@@ -376,4 +376,14 @@ public void testPartitionFilterWithLike() throws Exception {
     testIncludeFilter(query4, 4, "Filter", 16);
   }
 
+  @Test //DRILL-3710 Partition pruning should occur with varying IN-LIST size
+  public void testPartitionFilterWithInSubquery() throws Exception {
+    String query = String.format("select * from dfs_test.`%s/multilevel/parquet` where cast (dir0 as int) IN (1994, 1994, 1994, 1994, 1994, 1994)", TEST_RES_PATH);
+    /* In list size exceeds threshold - no partition pruning since predicate converted to join */
+    test("alter session set `planner.in_subquery_threshold` = 2");
```

Comment:
Not sure if it is necessary to check the no-partition-pruning case.  Basically, the goal of the test is to see if partition pruning works with large IN lists.